File: nestjs/src/business-rules/customers/handlers/delete-customer.handler.ts
import { IDeleteCustomerHandler } from './idelete-customer.handler';
import { Inject, Injectable, NotFoundException } from '@nestjs/common';
import {
  ICustomerRepository,
  ICustomerRepositoryName,
} from '@domain/repositories';
import { IDeleteCustomerRequest } from '../requests';

@Injectable()
export class DeleteCustomerHandler implements IDeleteCustomerHandler {
  constructor(
    @Inject(ICustomerRepositoryName) private repository: ICustomerRepository,
  ) {}

  async execute(request: IDeleteCustomerRequest): Promise<void> {
    const customer = await this.repository.findById(request.id);
    if (!customer) {
      throw new NotFoundException('Customer not found');
    }

    await this.repository.delete(request.id);
  }
}

File: dotnet/BusinessRules/Customers/Handlers/DeleteCustomerHandler.cs
using dotnet.BusinessRules.Customers.Requests;
using dotnet.BusinessRules.Exceptions;
using dotnet.Domain.Repositories;

namespace dotnet.BusinessRules.Customers.Handlers
{
    public class DeleteCustomerHandler : IDeleteCustomerHandler
    {
        private readonly ICustomerRepository _customerRepository;

        public DeleteCustomerHandler(
            ICustomerRepository customerRepository
        )
        {
            _customerRepository = customerRepository;
        }


        public void Execute(DeleteCustomerRequest request)
        {
            var customer = _customerRepository.FindById(request.Id);
            if (customer == null)
            {
                throw new NotFoundException("Cliente não encontrado");
            }

            _customerRepository.Delete(customer);
        }
    }
}